jena-arq/src/main/java/org/apache/jena/sparql/exec/http/Service.java

```diff
@@ -241,6 +242,37 @@ private static QuerySendMode chooseQuerySendMode(String serviceURL, Context cont
         return context.get(httpServiceSendMode, dftValue);
     }
 
+    private static QuerySendMode chooseQuerySendMode2(String serviceURL, Context context, QuerySendMode dftValue) {
+        if ( context == null )
+            return dftValue;
+        Object querySendMode = context.get(httpServiceSendMode, dftValue);
+        if ( querySendMode == null )
+            return dftValue;
+
+        if (querySendMode instanceof QuerySendMode)
+            // handle enum type from Java API
+            return (QuerySendMode) querySendMode;
+
+        if (querySendMode instanceof String) {
+            String str = (String) querySendMode;
+            // Specials.
+            if ( "POST".equalsIgnoreCase(str) )
+                return QuerySendMode.asPost;
+            if ( "GET".equalsIgnoreCase(str) )
+                return QuerySendMode.asGetAlways;
+            try {
+                // "asGetWithLimitForm", "asGetWithLimitBody", "asGetAlways", "asPostForm,asPost"
+                return QuerySendMode.valueOf((String) querySendMode);
+            } catch (IllegalArgumentException ex) {
+                FmtLog.warn(Service.class, "Failed to interrept '%s' as a query send mode - ignored", querySendMode);
```

Comment:
s/interrept/interpret

Reply:
Thanks for catching that. 

My bad - got my branches mixed up. This is not supposed to be in this PR.